### PR TITLE
test: add test scripts and functional tests for neutron chain, neutron-archway ibc, icon-neutron ibc

### PR DESCRIPTION
**Pull request by abhiyana**
### Commit Message

```bash
test: add test scripts and functional tests for neutron chain, neutron-archway ibc, icon-neutron ibc
```

see the [guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages) for commit messages.

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have documented my code in accordance with the [documentation guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#documentation)
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have run the unit tests
- [ ] I only have one commit (if not, squash them into one commit).
- [x] I have a descriptive commit message that adheres to the [commit message guidelines](https://github.com/icon-project/community/blob/main/guidelines/technical/software-development-guidelines.md#commit-messages)

> Please review the [CONTRIBUTING.md](/CONTRIBUTING.md) file for detailed contributing guidelines.

### CODE REVIEW

File: test/functional/dive_test.go [hunk 1]
```diff
@@ -273,6 +273,64 @@ var _ = ginkgo.Describe("DIVE CLI App", func() {
...
+
+		ginkgo.It("should start bridge between 2 custom neutron chains", func() {
+			dive.RunCustomNeutronNode0()
+			dive.RunCustomNeutronNode1()
+			cmd.Args = append(cmd.Args, "bridge", "ibc", "--chainA", "neutron", "--chainB", "neutron", "--chainAServiceName", "neutron-node-test-chain2", "--chainBServiceName", "neutron-node-test-chain2")
+			err := cmd.Run()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("should start bridge between 1 custom chain and running bridge command", func() {
```
Comment (hemz10) [inline test/functional/dive_test.go:292]: Please mention chain name

File: test/functional/dive_test.go [hunk 2]
```diff
@@ -273,6 +273,64 @@ var _ = ginkgo.Describe("DIVE CLI App", func() {
...
+
+		ginkgo.It("should start bridge between already running archway and neutron chains with custom configuration", func() {
+			dive.RunCustomNeutronNode0()
+			dive.RunCustomArchwayNode0()
+			cmd.Args = append(cmd.Args, "bridge", "ibc", "--chainA", "archway", "--chainB", "neutron", "--chainAServiceName", "node-service-archway-node-0", "--chainBServiceName", "neutron-node-test-chain2")
+			err := cmd.Run()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("should start IBC relay between icon and achway", func() {
```
Comment (hemz10) [inline test/functional/dive_test.go:322]: Common description for all test cases

File: test/functional/dive_test.go [hunk 3]
```diff
@@ -535,4 +593,39 @@ var _ = ginkgo.Describe("DIVE CLI App", func() {
 			gomega.Expect(err).NotTo(gomega.HaveOccurred())
 		})
 	})
+
+	ginkgo.Describe("Neutron chain commands", func() {
+		ginkgo.It("should run single neutron node", func() {
+			dive.RunNeutronNode()
+		})
+
+		ginkgo.It("should run single nurtron node with verbose flag enabled", func() {
```
Comment (hemz10) [inline test/functional/dive_test.go:602]: Typo

File: test/functional/dive_test.go [hunk 4]
```diff
@@ -535,4 +593,39 @@ var _ = ginkgo.Describe("DIVE CLI App", func() {
...
+			cmd.Args = append(cmd.Args, "chain", "neutron", "--verbose")
+			err := cmd.Run()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("should run single custom neutron node", func() {
```
Comment (hemz10) [inline test/functional/dive_test.go:608]: Add assertion

File: test/functional/dive_test.go [hunk 5]
```diff
@@ -535,4 +593,39 @@ var _ = ginkgo.Describe("DIVE CLI App", func() {
...
+			cmd.Args = append(cmd.Args, "chain", "neutron", "-c", "../../cli/sample-jsons/neutron5.json")
+			err := cmd.Run()
+			gomega.Expect(err).To(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("should output user that chain is already running when trying to run neutron chain that is already running", func() {
+			dive.RunNeutronNode()
+			cmd.Args = append(cmd.Args, "chain", "neutron")
+			err := cmd.Run()
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
```
Comment (hemz10) [inline test/functional/dive_test.go:628]: Error to be expected

File: test/scripts/cosmos/icon_neutron_e2e_demo.ts [hunk 6]
```diff
@@ -0,0 +1,238 @@
+import {
```
Comment (hemz10) [inline test/scripts/cosmos/icon_neutron_e2e_demo.ts:1]: Please make the cosmos test generic by having one ts file for all cosmos chain

**Review by shreyasbhat0 — APPROVED** [review]

**Pull request merged**